src/auth/ntlm/SSPI/ntlm_sspi_auth.cc

```diff
@@ -573,20 +572,20 @@ manage_request()
             /* check against SSPI */
             int err = ntlm_check_auth((ntlm_authenticate *) decoded, user, domain, decodedLen);
             have_challenge = 0;
-            if (err != NTLM_ERR_NONE) {
+            if (err != NtlmError::None) {
```

Comment:
C++11 [says](https://en.cppreference.com/w/cpp/language/enum) that "There are no implicit conversions from the values of a scoped enumerator to integral types". This is one of the primary differences between scoped and unscoped enumerators (by C++ design). AFAICT, this code requires such a conversion. My crude [example](https://cpp.godbolt.org/z/x5sGhu) fails to compile, as expected:

```
error: no match for 'operator!=' (operand types are 'int' and 'NtlmError')
     if (err != NtlmError::None)
```

I trust you have compiled this code. Can you explain why it works?

Reply:
I have now understood: that helper has bitrotten, and it doesn't build.
I've tried building it with Wine libraries, but it fails with lots of redefined #defines in the socket layer, but also argument type errors

Reply:
this specific issue can be fixed easily, as err is forcefully declared above but it's actually an NtlmError

Reply:
auto type deduction fixes it. Resolved.